flux-middle/src/rty/expr.rs

```diff
@@ -380,24 +469,30 @@ impl Expr {
 
         impl TypeFolder for Simplify {
             fn fold_expr(&mut self, expr: &Expr) -> Expr {
+                let span = expr.span();
                 match expr.kind() {
                     ExprKind::BinaryOp(op, e1, e2) => {
                         let e1 = e1.fold_with(self);
                         let e2 = e2.fold_with(self);
+                        let e1_span = e1.span();
+                        let e2_span = e2.span();
                         match (op, e1.kind(), e2.kind()) {
-                            (BinOp::And, ExprKind::Constant(Constant::Bool(false)), _)
-                            | (BinOp::And, _, ExprKind::Constant(Constant::Bool(false))) => {
-                                Expr::constant(Constant::Bool(false))
+                            (BinOp::And, ExprKind::Constant(Constant::Bool(false)), _) => {
+                                Expr::constant_at(Constant::Bool(false), e1_span)
+                            }
+                            (BinOp::And, _, ExprKind::Constant(Constant::Bool(false))) => {
+                                Expr::constant_at(Constant::Bool(false), e2_span)
                             }
                             (BinOp::And, ExprKind::Constant(Constant::Bool(true)), _) => e2,
                             (BinOp::And, _, ExprKind::Constant(Constant::Bool(true))) => e1,
                             (op, ExprKind::Constant(c1), ExprKind::Constant(c2)) => {
+                                let e2_span = e2.span();
                                 match Expr::const_op(op, c1, c2) {
-                                    Some(c) => Expr::constant(c),
-                                    None => Expr::binary_op(*op, e1, e2),
+                                    Some(c) => Expr::constant_at(c, span.or(e2_span)),
+                                    None => Expr::binary_op(*op, e1, e2, None),
```

Comment:
I think we should keep the original `expr.span()` here instead of `e2.span()`. This may be the reason we were blaming only the second operand in one of the examples

```
x > 0
    ^
```

The same for simplifying `false && e` and `e && false`